src/components/SchemaViewer.svelte

```diff
@@ -11,15 +11,32 @@
       .split(" ")
       .filter((t) => t.length > 0);
 
-    const addVisibility = (node) => {
+    const addVisibility = (node, parentNodeNames) => {
       let modifiedNode = node;
+      const parentNames =
+        parentNodeNames.length > 0 ? [...parentNodeNames] : [];
 
       if (modifiedNode.fields) {
-        modifiedNode.fields.forEach(addVisibility);
+        if (parentNames.length > 0) {
+          parentNames.push(`${parentNames.slice(-1)}.${modifiedNode.name}`);
+        } else {
+          parentNames.push(`${modifiedNode.name}`);
+        }
+
+        modifiedNode.fields.forEach((field) => {
+          let modifiedNodeField = field;
+          modifiedNodeField.parentNames = parentNames;
```

Comment:
We can have same results without resorting to side effects on the node.

Reply:
@acmiyaguchi Please can you elaborate on this? not sure I understand

Reply:
So it's not actually a [side-effect](https://en.wikipedia.org/wiki/Side_effect_(computer_science)) since the modified node is never being returned, but the idea is that the variable should be used directly instead of being set in the object.

Reply:
And just to be clear, my suggestion is to have removed this line.

Reply:
Removing this line flaws the entire logic as parentNames is needed